Add note about squash commits (#3721)
astronomer/docs: astro/ci-cd-templates/aws-s3.md

    - **Suffix**: Enter `.py`.

9. If you haven't already, deploy your complete Astro project to your Deployment. See [Deploy code](deploy-code.md).

>     :::tip

Reviewer: ```suggestion
    :::info
```
This seems more like a mishap to avoid rather than a tip to take advantage of

>     When you stage multiple commits locally that include DAG changes, and then push them all to your remote branch simultaneously, the DAG deploy template only deploys DAG code changes from the most recent commit. It might miss any code changes made in previous commits.

Reviewer: ```suggestion
    If you stage multiple commits to DAG files and push them all at once to your remote branch, the template only deploys DAG code changes from the most recent commit. It will miss any code changes made in previous commits.
```

>     Either push commits individually or configure your repository to **Squash commits** for pull requests that merge multiple commits simultanouesly.

Reviewer: ```suggestion
    To avoid this, either push commits individually or configure your repository to **Squash commits** for pull requests that merge multiple commits simultaneously.
```
